source/_UIAHandler.py

```diff
@@ -538,6 +538,25 @@ def _isUIAWindowHelper(self,hwnd):
 		# allow the appModule for the window to also choose if this window is bad
 		if appModule and appModule.isBadUIAWindow(hwnd):
 			return False
+		if windowClass == "NetUIHWND":
+			# NetUIHWND is used for various controls in MS Office.
+			# IAccessible should be used for NetUIHWND in versions older than 2016
+			# Fixes: lack of focus reporting (#4207),
+			# Fixes: strange reporting of context menu items(#9252),
+			# fixes: not being able to report ribbon sections when they starts with an edit  field (#7067)
+			# Note that #7067 is not fixed for Office 2016 and never.
+			# Using IAccessible for NetUIHWND controls causes focus changes not to be reported
+			# when the ribbon is collapsed.
+			# Testing shows that these controls emits proper events but they are ignored by NVDA.
+			if(
+				appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
+				and int(appModule.productVersion.split(".")[0]) < 16
+			):
+				parentHwnd = winUser.getAncestor(hwnd, winUser.GA_PARENT)
+				while parentHwnd:
+					if winUser.getClassName(parentHwnd) in ("Net UI Tool Window", "MsoCommandBar",):
+						return False
+					parentHwnd = winUser.getAncestor(parentHwnd, winUser.GA_PARENT)

```

Comment:
App module might be None. The following suggestion makes this check when matching the window class name, and names the checks for `isOfficeApp` and `isBefore2016Version`
``` python
		if windowClass == "NetUIHWND" and appModule:
			# NetUIHWND is used for various controls in MS Office.
			# IAccessible should be used for NetUIHWND in versions older than 2016
			# Fixes: lack of focus reporting (#4207),
			# Fixes: strange reporting of context menu items(#9252),
			# fixes: not being able to report ribbon sections when they starts with an edit  field (#7067)
			# Note that #7067 is not fixed for Office 2016 and newer.
			# Using IAccessible for NetUIHWND controls causes focus changes not to be reported
			# when the ribbon is collapsed.
			# Testing shows that these controls emits proper events but they are ignored by NVDA.
			isOfficeApp = appModule.productName.startswith(("Microsoft Office", "Microsoft Outlook"))
			isBefore2016Version =  int(appModule.productVersion.split(".")[0]) < 16
			if isOfficeApp and isBefore2016Version:
				parentHwnd = winUser.getAncestor(hwnd, winUser.GA_PARENT)
				while parentHwnd:
					if winUser.getClassName(parentHwnd) in ("Net UI Tool Window", "MsoCommandBar",):
						return False
					parentHwnd = winUser.getAncestor(parentHwnd, winUser.GA_PARENT)

```